addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -12803,4 +12803,31 @@ QUnit.module("Views", (hooks) => {
         await click(target.querySelector(".o_data_row .o_data_cell"));
         assert.containsOnce(target, ".o_form_view");
     });
+
+     QUnit.test("archive action not shown with readonly active field", async function (assert) {
+
+        // add active field on partner model, but do not put it in the view
+        serverData.models.partner.fields.active = {
+            string: "Active",
+            type: "char",
+            default: true,
+            readonly: true,
+        };
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            resId: 1,
+            serverData,
+            arch: `<form><field name="foo"/></form>`,
+            actionMenus: {},
+        });
+        await click(target, ".o_cp_action_menus .dropdown-toggle");
+        assert.deepEqual(
+            [...target.querySelectorAll(".o_menu_item")].map((el) => el.textContent),
+         ["Duplicate", "Delete"],
+            "Should not contain an Archive action",
+
+        );
+
+    });
```

Comment:
In order to keep things a little tidy, move this new test right after the one named "archive action with active field not in view".